src/nvim/os/input.c

```diff
@@ -118,7 +115,8 @@ int os_inchar(uint8_t *buf, int maxlen, int ms, int tb_change_cnt, MultiQueue *e
       return 0;
     }
   } else {
-    if ((result = inbuf_poll((int)p_ut, events)) == kInputNone) {
+    if ((result = inbuf_poll((int)p_ut, events)) == kInputNone
+        || !typebuf_changed(tb_change_cnt)) {
```

Comment:
If events are available this can trigger `CursorHold` immediately.

Reply:
Well, it is hard to wait `updatetime` properly.